src/components/wizard/wizard.component.ts

```diff
@@ -178,27 +183,66 @@ export class WizardComponent implements AfterViewInit, OnDestroy {
      */
     next(): void {
 
+        this.currentStep = this.getCurrentStep();
+
+        if (this.currentStep.validator) {
+
+            let validationResult = this.currentStep.validator();
+
+            const observable = typeof validationResult === 'boolean' || validationResult === null ? of(validationResult) : from(validationResult); // doesnt accept null
+
+            this.subscription = observable.pipe(first()).subscribe(
+                isValid => {
+
+                    this.currentStep.valid = isValid;
+                },
+                err => {
+                    console.error('ux-wizard: Validator function has thrown an error');
+                    console.error(err);
+                },
+                () => {
+                    this.wizardValidationCompleted();
+
+                }
+            );
+
+            // only add the subscription to the list of requests if it isnt complete.
+            if (!this.completed) {
+                this._subscriptions.push(this.subscription);
+            }
+
+        } else {
+            this.wizardValidationCompleted();
+        }
+    }
+
+    wizardValidationCompleted() {
```

Comment:
This function is really responsible for the next logic and it should probably remain `next`. We also might want to run validation on `finish`, in which case this would involve duplicating all the validation logic. It would be better to have validation in a separate function that the next function calls, and if it returns a promise then we can use `await` to wait until any observable or promise has reolved, eg:

```typescript
async next(): void {

   if (!await this.isStepValid()) {
        return;
   }
...

```